docs/decisions/010-deprecations.md

```diff
@@ -0,0 +1,59 @@
+# Deprecations
+
+Date: 2024-11-18
+
+Status: proposed
+
+## Context
+
+Starting the development of our design system from scratch, we have been using version 0 for a long time.
+As [SEMVER suggests][semver-zero-version], we want to communicate to our users that the design system is at an early stage and any feature is not considered stable.
+However, from the time our users started to use the design system and test it, they wanted the components that would not change under the hands.
+They want to API to be more stable and predictable.
+
+Starting the first production usage of the design system, we had to stabilize the API and find another way how to customize and further develop our components.
+But without breaking the existing API.
+
+So, we began with the research on how other libraries are solving this problem.
+In various libraries like ESLint or [Twig][twig-deprecated] we have found the concept of deprecations.
+
+This suits well to our needs, because we can mark the features we want to change or remove in the future and give our users a clear path how to migrate to the new version.
+
+## Decision
+
+Every feature that will significantly change or be removed in the next major version must be marked as deprecated.
+
+We will implement the following deprecation strategy:
+
+1. All deprecations must use warning messages in development environments
+2. Warnings must be disabled in production environments
+3. Each deprecation warning must include:
+   - What is being deprecated
+   - When it will be removed (usually the next major version)
+   - What to use instead
+4. Deprecation warnings will be shown in the browser console or the terminal
+
+## Consequences
+
+We will mark any feature that will be considered as a breaking change as deprecated.
+We will maintain DEPRECATIONS-v\*.md files for each major version documenting all deprecations.
+We will include a deprecation notice in the commit message as well as in the release notes.
+We will provide [migration guides][migration-guides] for each deprecation.
+There will be a "Deprecations" section in each package README.md file pointing to the deprecations list.
+When there will be a deprecation notice in the component's README file it will be pointing to the related deprecation notice in deprecations list.
```

Comment:
```suggestion
When there will be a deprecation notice in the component's README file it will be pointing to the related deprecation notice in the deprecations list.
```

Reply:
Fixed.